operators/multiclusterobservability/controllers/multiclusterobservability/multiclusterobservability_controller.go

```diff
@@ -151,6 +151,28 @@ func (r *MultiClusterObservabilityReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, nil
 	}
 
+	if _, ok := config.BackupResourceMap[instance.Spec.StorageConfig.MetricObjectStorage.Key]; !ok {
+		config.BackupResourceMap[instance.Spec.StorageConfig.MetricObjectStorage.Key] = config.ResourceTypeSecret
+	}
+
+	if res, ok := config.BackupResourceMap[req.Name]; ok {
```

Comment:
seems a timing issue here.
Assume we don't create MCO CR firstly. And the create events for those backupresource will be ignored. Have you tested that scenario?

Reply:
Moved all processing ahead of MCO CR check to catch this case.